pkg/verification/verify.go

```diff
@@ -44,6 +166,36 @@ func VerifyTimestampResponse(tsrBytes []byte, artifact io.Reader, certPool *x509
 		return err
 	}
 
+	err = verifyNonce(ts.Nonce, opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyOID(ts.Policy, opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyEmbeddedLeafCert(ts.Certificates[0], opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyLeafAndIntermediatesEKU(opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyLeafCertSubject(ts.Certificates[0].Subject.String(), opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyESSCertID(ts.Certificates[0], opts)
```

Comment:
nit: Perhaps we can group the ts.Certificates[0] validation into a single function with all these sub-functions.